### ISSUE TITLE
Does the eval_metric choice affect training process, i.e., is metric related to loss?

### ISSUE THREAD

**Issue by 300LiterPropofol** [awslabs/autogluon#4312]
### Describe the issue linked to the documentation

link: https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-metrics.html

> AutoGluon will use the provided metric to tune model hyperparameters, rank models, and construct the final ensemble for prediction.

I am looking into the time series prediction part of Autogluon package. I am confused about the eval_metric choice here. It is used for the final evaluation of the trained model as its name self explains, but I wonder whether it is also working as the loss of the training process, i.e., the training back propagation process is aiming to make the eval_metric value minimum.
 
During my experiment, I found that if I train with the same train_data using the same model but with different eval_metric, and then evaluate the finished model using the same list of eval_metric, I will get very different results.

### Suggest a potential alternative/fix

It will be nice to confirm in the documentation how exactly the eval_metric is used.
Thank you!

**Comment by shchur** [comment]
Hi @300LiterPropofol, for most models the choice of the metric has no effect on the training procedure with the exception of `DirectTabular` and `RecursiveTabular`.  These sections of the tutorial ([1](https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-metrics.html#custom-mean-absolute-scaled-error-metric), [2](https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-metrics.html#custom-quantile-loss-metric)) explain how different properties of the `eval_metric` affect the configuration of the DirectTabular and RecursiveTabular models.

**Comment by 300LiterPropofol** [comment]
@shchur Thank you for your explanation!
I am only using the timeSeries part so I am not using the two models you pointed out. Since I noticed that using the same train_data and the same model (for example tenporalfusiontranformer), but using different eval_metric (SQL and MAE) to configure the predictor, after training I evaluate the model using the same test_data on the same MAE metric, the evaluation MAE has obvious differences ( I understand multiple training will results in multiple different parameter sets so I did multiple times to test but the difference is obvious regardless). The documentation mentioned eval_metric affects hyper parameter tuning. Can I think this difference as purely because of hyperparameter tuning streátegy difference since as you mentioned the training loss is not related to eval_metric.

**Comment by Innixma** [comment]
@300LiterPropofol The eval_metric will determine the weighted ensemble's logic for how much weight to give to each model to maximize the validation score of the eval_metric. This can then lead to differences in scores.

**Comment by shchur** [comment]
@300LiterPropofol, yes, you are absolutely right. The hyperparameter tuning may select very different hyperparameter configurations if you change the evaluation metric, which will lead to different predictions.

**Comment by shchur** [comment]
This is now clarified in the documentation https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-metrics.html#customizing-the-training-loss-for-individual-models.
